cfg/config.go

```diff
@@ -0,0 +1,184 @@
+/*
+ * knoxite
+ *     Copyright (c) 2020, Nicolas Martin <penguwin@penguwin.eu>
+ *
+ *   For license see LICENSE
+ */
+package cfg
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	gap "github.com/muesli/go-app-paths"
+)
+
+const appName = "knoxite"
+
+var cfgFileName = "knoxite.conf"
+
+// The RepoConfig struct contains all the default values for a a repository
+type RepoConfig struct {
+	Url         string `json:"url"`
+	Compression string `json:"compression"`
+	Tolerance   uint   `json:"tolerance"`
+	Encryption  string `json:"encryption"`
+}
+
+type Config struct {
+	Repositories map[string]RepoConfig
```

Comment:
Is this stored to json? If so, i would add an ``json:"repositories"``struct tag here - I think json-keys should not start with upper keys.

Reply:
Forgot the tag here - fixed it